src/MUSI6106Exec/tests.h

```diff
@@ -0,0 +1,180 @@
+    assert (test3error == Error_t::kNoError);
+    return test3error;
+}
+
+Error_t test4(CCombFilterIf::CombFilterType_t filterType)
+{
+//    std::string sInputFilePath = "/Users/vedant/Desktop/Programming/2022-MUSI6106/audio_files/silence.wav";
+//    std::string sOutputFilePath = "/Users/vedant/Desktop/Programming/2022-MUSI6106/audio_files/output/silence_filtered.wav";
+//    std::string filterName;
+//    if (filterType == CCombFilterIf::kCombIIR)
+//    {
+//        filterName = "IIR";
+//    }
+//    else if (filterType == CCombFilterIf::kCombFIR)
+//    {
+//        filterName = "FIR";
+//    }
+//    float gain = 0.5;
+//    float delay = 0.01;
+//    int blockSize = 1024;
+//
+//    Error_t test4error = run_filtering(sInputFilePath,sOutputFilePath,filterType,gain,delay,blockSize);
+    int iNumChannels = 2;
+    int kBlockSize = 1024;
+    int iNumFrames = kBlockSize;
+    float **ppfInputBuffer = nullptr;
+    float **ppfOutputBuffer = nullptr;
+
+    CCombFilterIf *filter = nullptr;
+    CCombFilterIf::create(filter);
+    filter->init(filterType,0.01,44100,iNumChannels, 0.5);
+    ppfInputBuffer = new float*[iNumChannels];
+    for (int i = 0; i < iNumChannels; i++)
+    {
+        ppfInputBuffer[i] = new float[kBlockSize];
+    }
+    ppfOutputBuffer = new float*[iNumChannels];
+    for (int i = 0; i < iNumChannels; i++)
+    {
+        ppfOutputBuffer[i] = new float[kBlockSize];
+    }
+
+    //Set input buffer to zeros
+    for (int c = 0; c < iNumChannels; c++)
+    {
+        for(int i = 0; i < iNumFrames;i++)
+        {
+            ppfInputBuffer[c][i] = 0;
+        }
+    }
+    //Process
+    Error_t test4 = filter->process(ppfInputBuffer,ppfOutputBuffer,iNumFrames);
+    //Clear memory
+    CCombFilterIf::destroy(filter);
+    for (int i = 0; i < iNumChannels; i++)
+    {
+        delete[] ppfInputBuffer[i];
+        delete[] ppfOutputBuffer[i];
+    }
+    delete[] ppfInputBuffer;
+    delete[] ppfOutputBuffer;
+    ppfInputBuffer = nullptr;
+    ppfOutputBuffer = nullptr;
+    assert (test4 == Error_t::kNoError);
+    cout << "Test with silence successful!" << endl;
+    return test4;
+}
+
+Error_t test5(CCombFilterIf::CombFilterType_t filterType)
+{
+    std::string sInputFilePath = "/Users/vedant/Desktop/Programming/2022-MUSI6106/audio_files/6.wav";
+    std::string sOutputFilePath = "/Users/vedant/Desktop/Programming/2022-MUSI6106/audio_files/output/6_filtered.wav";
+    std::string filterName;
+    if (filterType == CCombFilterIf::kCombIIR)
+    {
+        filterName = "IIR";
+    }
+    else if (filterType == CCombFilterIf::kCombFIR)
+    {
+        filterName = "FIR";
+    }
+    int blockSize = 1024;
+    float gain = -0.25f;
+    float delay = -0.125f;
+    Error_t test5error = run_filtering(sInputFilePath,sOutputFilePath,filterType,gain,delay,blockSize);
+    assert ((test5error==Error_t::kNoError) || (test5error==Error_t::kFunctionInvalidArgsError));
```

Comment:
Why the `||` here? It looks like you mean test that there actually was an error, but this assert will pass even if there wasn't because of the left half.